src/script/main/app.ts

```diff
@@ -613,6 +614,19 @@ class App {
     }
 
     await container.resolve(StorageService).init(this.core.storage);
+
+    //storage was initialised, possible version upgrade, we can try running migration
+    const dbMigrationState = dbMigrationStateStore.getDBMigrationState();
+
+    if (dbMigrationState) {
+      try {
+        await this.core.service?.mls.proteusCryptoboxMigrate(dbMigrationState.storeName);
```

Comment:
I feel like it is the best place to trigger migration script - right after storage has been initialised - we're sure that it's filled with data (and version upgrade hook did run)